Support for Amazon MQ Rabbit MQ queues

liegeandlief opened the pull request:
@pgrzesik This should add support for Amazon MQ RabbitMQ events. It is very much based upon the recently merged PR for Active MQ events. I have added integration tests but have been unable to actually run them as I don't have access to an AWS account with the necessary permissions. If you could take a look then that would be great. Thanks :)

liegeandlief commented:
Closes: https://github.com/serverless/serverless/issues/9924

pgrzesik commented:
Hello @liegeandlief - I've tried it out and after fixing few minor issues in integration tests they are failing, looks like the Lambda function cannot access rabbitmq at all. Did you have a chance to run the tests itself or did you perform similar testing to ensure it's working?

liegeandlief commented:
Hi @pgrzesik, thanks for taking a look. Is that the Lambda function for putting a message into the queue? Unfortunately I have not been able to run the integration tests as I don't have access to an AWS account with suitable permissions to create all the necessary infrastructure. Is there a particular error you are getting?

pgrzesik commented:
Hey @liegeandlief - yes, this it's the producer Lambda that cannot access the broker, it errors out with `"errorMessage":"read ECONNRESET"` - I'm guessing it might be caused by misconfigured SSL or something similar, I'm not really familiar with the `amqplib`

liegeandlief commented:
@pgrzesik There was a spelling mistake in one of the variable names which may have caused this error. I've have just pushed up a fix.

pgrzesik commented:
> @pgrzesik There was a spelling mistake in one of the variable names which may have caused this error. I've have just pushed up a fix.

Yeah, that one I've spotted and fixed on my side as it was simply trying to connect to localhost, but after fixing it it's still refusing to connect :(

liegeandlief commented:
@pgrzesik I'm just trying to build the infrastructure stack so I can take a look at this test error. I have fixed the spelling mistake problem and also corrected the RabbitMQ instance type to `mq.t3.micro`. But the `AWS::MSK::Cluster` seems to be stuck in the `CREATE_IN_PROGRESS` state. Did you encounter this or have to make any other changes to get the dependency stack running?

pgrzesik commented:
Hey @liegeandlief, the MSK Cluster takes really long to provision, but I would encourage you to just simplify the script to not provision `ActiveMQ`-related, `EFS-related` and `MSK`-related resources - just stuff that is needed for `RabbitMQ` - it took 10-15 minutes for me to provision such simplified stack (I simply commented a bunch of stuff in the template and script that provisions the infra).

liegeandlief commented:
@pgrzesik The RabbitMQ integration test should now be working (at least i worked for me!)

pgrzesik commented:
Thanks @liegeandlief - I see that it was about SSL in the end :D I've tried to test, but I see the integration test failing for me - there are not messages to be consumed by the consumer function that include the expected message. Is that test green for you?

liegeandlief commented:
@pgrzesik It's weird that it isn't working for you. It is green for me.
<img width="1251" alt="Screenshot 2021-09-13 at 17 13 33" src="https://user-images.githubusercontent.com/12057877/133119690-3a171758-4355-42cd-93a6-c577b42c79cc.png">

pgrzesik commented:
@liegeandlief That is surprising - for me it always fail with the following output

![rabbitmqfailing](https://user-images.githubusercontent.com/17499590/133247742-38da244b-a075-48d9-be95-7c17ba70536e.png)

liegeandlief commented:
@pgrzesik No local changes, everything is pushed

liegeandlief commented:
@pgrzesik When the test is running can you see the consumer function log group in CloudWatch and are you able to inspect what gets logged?

liegeandlief commented:
@pgrzesik a colleague and I have both managed to run the test successfully. We both used brand new AWS accounts when doing so and all the resources were created in the us-east-1 region. The AWS user credentials had full permissions on each account. When creating the integration test dependency stack we didn't comment anything out, but allowed it to create all infrastructure even though only some of it was required for the test. I don't know if any of this context is helpful in determining why the test is failing for you?

pgrzesik commented:
Hello @liegeandlief - I was trying it out once again today multiple times, with the exact same environment and the tests are not succeeding for me. Unfortunately, I won't be available for the next two weeks - I'm handing off the tasks to @medikoo, who based on availability might be able to help.

pgrzesik commented:
Hello @liegeandlief :wave: Sorry for the delay but as I've mentioned I've been off for two weeks. I'll be checking your branch and tests out once again on a fully fresh environment to see if I can get these tests to pass on my side. Thanks for you patience on this one :bow:

liegeandlief commented:
@pgrzesik I've brought my fork up to date with the upstream master branch
